test/test_performance.py.in

```diff
@@ -66,7 +66,7 @@ def _raw_to_csv(csv_data, csv_path):
          csv.write(','.join(values) + '\n')
 
 def _raw_to_png(csv_data, png_path):
```

Comment:
All the plots generated here have different axis ranges extrapolated from the data, which makes visual comparisons very difficult.

Reply:
@ahcorde @cottsay I'd love to hear your thoughts about this. I was thinking that another possibility would be to save the dataframe with the raw data to a CSV file (keeping the original column names) so that users can look at the numbers and create their own graphs

Reply:
This is a rectriction from the Jenkins plot plugin, This column is used to show the legend.

Reply:
I think saving the CSV for other uses sounds fine to me.

We might be reaching the point where we need to separate the date collection from the presentation. If we had all of the CSV data saved in a discoverable location with some predefined naming scheme, we could create an aggregation script. Since that aggregation script would have context from all of the plots collectively, we could do things like enforce common axis domain.

The main reason that this is difficult to integrate into the buildfarm infrastructure is that right now, the performance test jobs are just a normal CI build. We don't have a way to just run an arbitrary script at the end of the test run, so we need all of the operations to be encapsulated in a test of some sort. Passing files between tests arbitrarily seems pretty fragile to me, so we'd need to do some experimentation to find the best way to make it reliable and maintainable.

It's do-able, but it's out of scope here. A follow-up PR to save the CSVs would be appropriate, though.

Reply:
Filed as #30